lib/variant-model.js

```diff
@@ -45,17 +47,39 @@ _.extend(Variant.prototype, {
     return this;
   },
 
-  respondWithFile: function(fileName) {
+  respondWithFile: function(options) {
+    var self = this;
+
     return this.respondWith(function(request, reply) {
-      var self = this;
-      fileName = fileName.replace(/\{([^\}]+)\}/g, function(match, token) {
-        var val = request.params[token];
-        if (!val) {
-          val = self.state(token);
+      // Intentionally requiring smocks to lazy load options
+      var smocks = require('./index');
+
+      options = _.extend({
+        fileName: smocks.initOptions.fileName || 'default.json',
+        mockDir: smocks.initOptions.mockDir || './mocked-data',
+        statusCode: smocks.initOptions.statusCode || 200
+      }, options);
+
+      var pattern = /{\w+}\/*/g;
+      var path = self._route.path().replace(pattern, '');
+      var dataVariantFilePath = Path.join(options.mockDir, path, self.id() + '.json');
+
+      // Search for variant file
+      Fs.readFile(dataVariantFilePath, function (err, variantData) {
+        if (err) {
+          // No variant file found, search for the default json file
+          var dataDefaultFilePath = Path.join(options.mockDir, path, options.fileName);
+          Fs.readFile(dataDefaultFilePath, function (err, defaultData) {
+            if (err) {
+              return reply().code(404);
+            } else {
+              reply(defaultData).type('application/json').code(options.statusCode);
+            }
+          });
+        } else {
+          reply(variantData).type('application/json').code(options.statusCode);
         }
```

Comment:
Hmm... not sure I agree.  The whole point of a variant is that you would want a different response for some reason - what would the use case be where you would want a variant with the same response as default?